Test/CoreSDK.Test/Shared/Extensibility/Implementation/TaskTimerInternalTest.cs

```diff
@@ -203,6 +210,34 @@ public void AbortsPreviousAction()
                     Assert.False(actionInvoked);
                 });
             }
+
+            [TestMethod]
+            public void CancelAbortsOnlyLastTask()
+            {
+                var timer = new TaskTimerInternal { Delay = TimeSpan.FromMilliseconds(25) };
+
+                int invocationCount = 0;
+                var firstActionInvoked = new ManualResetEventSlim();
+                var secondActionInvoked = new ManualResetEventSlim();
+                timer.Start(() => Task.Factory.StartNew(
+                    () =>
+                    {
+                        Interlocked.Increment(ref invocationCount);
+                        firstActionInvoked.Set();
+                    }));
+                timer.Start(
+                    () => Task.Factory.StartNew(
+                        () =>
+                        {
+                            Interlocked.Increment(ref invocationCount);
+                            secondActionInvoked.Set();
+                        }));
+
+                timer.Cancel();
```

Comment:
Cancel should actually take the token to cancel instead of just cancelling the most recent.